File: ColdDream.Api/Models/Guide.cs
using System.ComponentModel.DataAnnotations;

namespace ColdDream.Api.Models;

public class Guide
{
    public Guid Id { get; set; }
    public Guid UserId { get; set; }
    public string Title { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public string? ImageUrl { get; set; }
    public string? Itinerary { get; set; } // JSON string: [{day: 1, title: "", content: ""}]
    public int Status { get; set; } = 0; // 0: Pending, 1: Published, 2: Rejected
    public string? Tags { get; set; } // JSON array: ["Tag1", "Tag2"]
    public string? Duration { get; set; }
    public string? Budget { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
}

File: ColdDream.UniApp/src/api/guide.ts
import { request, type ApiResponse } from '@/utils/request';

export interface Guide {
    id: string;
    userId: string;
    title: string;
    description: string;
    imageUrl?: string;
    itinerary?: string;
    status: number; // 0: Pending, 1: Published, 2: Rejected
    tags?: string; // JSON array string
    duration?: string;
    budget?: string;
    createdAt: string;
    updatedAt: string;
}

export const getMyGuides = () => {
    return request<ApiResponse<Guide[]>>({
        url: '/guide/my',
        method: 'GET'
    });
};

export const getGuideById = (id: string) => {
    return request<ApiResponse<Guide>>({
        url: `/guide/${id}`,
        method: 'GET'
    });
};

export const createGuide = (data: Partial<Guide>) => {
    return request<ApiResponse<Guide>>({
        url: '/guide',
        method: 'POST',
        data
    });
};

export const updateGuide = (id: string, data: Partial<Guide>) => {
    return request<ApiResponse<any>>({
        url: `/guide/${id}`,
        method: 'PUT',
        data
    });
};

export const deleteGuide = (id: string) => {
    return request<ApiResponse<any>>({
        url: `/guide/${id}`,
        method: 'DELETE'
    });
};
